File: rules-interpreter/src/main/java/io/github/iamrenny/ruleflow/evaluators/ContextEvaluator.java
package io.github.iamrenny.ruleflow.evaluators;

import io.github.iamrenny.ruleflow.errors.PropertyNotFoundException;
import io.github.iamrenny.ruleflow.errors.UnexpectedSymbolException;
import io.github.iamrenny.ruleflow.visitors.Visitor;
import org.antlr.v4.runtime.ParserRuleContext;

/**
 * Interface for context evaluators in the RuleFlow engine.
 * Implementations should provide logic for evaluating specific parse tree contexts.
 */
public interface ContextEvaluator<T extends ParserRuleContext> {
    Object evaluate(T ctx, Visitor visitor)
        throws PropertyNotFoundException, UnexpectedSymbolException;
}

File: rules-interpreter/src/main/kotlin/com/rappi/analang/evaluators/ContextEvaluator.kt
package com.github.iamrenny.rulesflow.evaluators

import com.github.iamrenny.rulesflow.visitors.Visitor
import org.antlr.v4.runtime.ParserRuleContext
import org.antlr.v4.runtime.RuleContext

interface ContextEvaluator<T: ParserRuleContext> {

    fun evaluate(ctx: T, visitor: Visitor): Any
}